registry/resumable: TestResumableRequestHeaderTooMuchFailures fails with Go 1.14 beta1

Fedora Rawhide with Go 1.14 beta1, Docker Engine v19.03.5

```
--- FAIL: TestResumableRequestHeaderTooMuchFailures (0.00s)
    resumablerequestreader_test.go:80: assertion failed: expected error "Get I%27m%20not%20an%20url: unsupported protocol scheme \"\"", got "Get \"I%27m%20not%20an%20url\": unsupported protocol scheme \"\""
time="2020-01-30T01:23:53Z" level=info msg="encountered error during pull and clearing it before resume: An error occurred"
FAIL
```

See golang/go@64cfe9f
and
golang/go#29261

--- a/registry/resumable/resumablerequestreader_test.go
+++ b/registry/resumable/resumablerequestreader_test.go
@@ -75,9 +75,10 @@ func TestResumableRequestHeaderTooMuchFailures(t *testing.T) {
 	}
 	defer resreq.Close()
 
-	expectedError := `Get I%27m%20not%20an%20url: unsupported protocol scheme ""`
 	read, err := resreq.Read([]byte{})
-	assert.Check(t, is.Error(err, expectedError))
+	assert.Assert(t, err != nil)
+	assert.Check(t, is.ErrorContains(err, "unsupported protocol scheme"))
+	assert.Check(t, is.ErrorContains(err, "I%27m%20not%20an%20url"))
 	assert.Check(t, is.Equal(0, read))
 }
 

